### PR TITLE
Group vector queries into new section

### PR DESCRIPTION
**Pull request by carlosdelest**
Group together vector queries in a new Query DSL section. Moves vector queries from the specialized queries section.

This helps creating a base page for vector queries, avoid the queries being scattered, and cleans up a bit the Query DSL section.

This is the new Query DSL layout:

image

### CODE REVIEW

File: docs/reference/query-dsl/vector-queries.asciidoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[[vector-queries]]
+== Vector queries
+
+Vector queries search efficiently for vectors in order to perform <<semantic-search,semantic search>>.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/vector-queries.asciidoc:4]: May be better phrase like this: "Vector queries are specialized queries that work on vector fields to efficiently perform semantic search"?

**Review by mayya-sharipova — APPROVED** [review]
@carlosdelest Thanks, makes sense

**Review by ChrisHegarty — APPROVED** [review]
LGTM

File: docs/reference/query-dsl/vector-queries.asciidoc (hunk 1 above)
Comment (carlosdelest) [reply docs/reference/query-dsl/vector-queries.asciidoc:4]: I like that, thanks!

**Review by leemthompo — APPROVED** [review]
This looks great @carlosdelest!

One thought was to perhaps group `text_expansion` and `weighted_tokens` into an appendix called something like "Legacy vector queries" to communicate visually that they have been superseded. WDYT?

File: docs/reference/query-dsl/vector-queries.asciidoc [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+<<query-dsl-knn-query,`knn` query>>::
+A query that finds the _k_ nearest vectors to a query vector for <<dense-vector,`dense_vector`>> fields, as measured by a similarity metric.
+
+<<query-dsl-sparse-vector-query,`sparse_vector` query>>::
+A query used to search <<sparse-vector,`sparse_vector`>> field types.
+
+<<query-dsl-semantic-query,`semantic` query>>::
+A query that allows you to perform semantic search on <<semantic-text,`semantic_text`>> fields.
+
+=== Deprecated vector queries
```
Comment (carlosdelest) [inline docs/reference/query-dsl/vector-queries.asciidoc:15]: @leemthompo , WDYT? I think adding an additional WARN is overkill, but double checking.

Comment (leemthompo) [reply docs/reference/query-dsl/vector-queries.asciidoc:15]: Note-perfect as always @carlosdelest 👍

File: docs/reference/query-dsl/vector-queries.asciidoc [hunk 3]
```diff
@@ -12,6 +12,7 @@ A query used to search <<sparse-vector,`sparse_vector`>> field types.
 <<query-dsl-semantic-query,`semantic` query>>::
 A query that allows you to perform semantic search on <<semantic-text,`semantic_text`>> fields.
 
+[discrete]
```
Comment (leemthompo) [inline docs/reference/query-dsl/vector-queries.asciidoc:15]: `::shakes fist::`

**Pull request merged**